### PR TITLE
rm listener & add sleep

### PR DESCRIPTION
**Pull request by javi555**

### CODE REVIEW

File: modules/services/queen_bee_svc/src/queen_bee_svc.cpp [hunk 1]
```diff
@@ -29,17 +35,23 @@ void QueenBeeSvc::processImg(const kpsr::vision_ocv::ImageData &img) {
...
   mls::Waypoint waypoint = calculateWaypointForImage(img);
-  //for loop in publishers     
-  _waypointPublisher->publish(waypoint);
-  spdlog::info("QueenBeeSvc: Published waypoint");
+  for (int i = 0; i < _waypointPublishers.size(); i++)
+  {
+    if (_waypointPublishers[i]._minIndex < img.seq < _waypointPublishers[i]._maxIndex){
+      _waypointPublishers[i]._waypointPublisher->publish(waypoint);
+      spdlog::info("QueenBeeSvc: Publisher_{} waypoint {}", i, waypoint.seq);
+    }
```
Comment (pablogh-2000) [inline modules/services/queen_bee_svc/src/queen_bee_svc.cpp:50]: addd ```break```

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp [hunk 2]
```diff
@@ -16,9 +16,8 @@ ZmqMiddlewareFacility::ZmqMiddlewareFacility(zmq::context_t &context,
     , _imageDataSubscriber(nullptr)
     , _imageDataPublisher(nullptr)
     , _waypointSubscriber(nullptr)
-    , _waypointPublisher(nullptr)
     , _binaryFromZMQProvider(nullptr)
-    , _toZMQPublisher(nullptr)
+    , _toZMQPublishers()
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp]: ????

File: modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h [hunk 3]
```diff
@@ -43,8 +43,8 @@ class ZmqMiddlewareFacility
     kpsr::Subscriber<kpsr::vision_ocv::ImageData> *_imageDataSubscriber;
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h:44]: ```std::vector<std::shared_ptr<zmq::socket_t>> _publishers;```

File: modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h [hunk 4]
```diff
@@ -43,8 +43,8 @@ class ZmqMiddlewareFacility
...
     kpsr::Publisher<kpsr::vision_ocv::ImageData> *_imageDataPublisher;
     kpsr::Subscriber<mls::Waypoint> *_waypointSubscriber;
-    kpsr::Publisher<mls::Waypoint> *_waypointPublisher;
     kpsr::zmq_mdlw::FromZmqChannel<Base> *_binaryFromZMQProvider;
+    std::vector<mls::WpPublisherConfig> _toZMQPublishers;
     kpsr::Publisher<mls::Waypoint> *_toZMQPublisher;
     kpsr::zmq_mdlw::FromZmqMiddlewareProvider _fromZmqMiddlewareProvider;
     kpsr::zmq_mdlw::ToZMQMiddlewareProvider _toZMQMiddlewareProvider;
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h]: ```std::vector<std::shared_ptr<psr::zmq_mdlw::ToZMQMiddlewareProvider _toZMQMiddlewareProvider>>;```

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp [hunk 5]
```diff
@@ -16,9 +16,8 @@ ZmqMiddlewareFacility::ZmqMiddlewareFacility(zmq::context_t &context,
...
     , _subscriber(context, ZMQ_SUB)
     , _publisher(context, ZMQ_PUB)
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp]: Add number of publishers ```int noPublishers```
```
_publishers(noPublishers),
_toZMQPublisher(noPublishers)
```

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp [hunk 6]
```diff
@@ -16,9 +16,8 @@ ZmqMiddlewareFacility::ZmqMiddlewareFacility(zmq::context_t &context,
...
     , _fromZmqMiddlewareProvider()
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp:21]: Add loop inside constructor:
```
_publishers[i] = new zmq::socket_t(context, ZMQ_PUB);
_toZmqPublishers[i] = new ToZMQChannel(_publisher[i].get(), etc)
```

File: modules/services/queen_bee_svc/src/queen_bee_svc.cpp (hunk 1 above)
Comment (javi555) [reply modules/services/queen_bee_svc/src/queen_bee_svc.cpp:50]: Done

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp (hunk 6 above)
Comment (javi555) [reply modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp:21]: Done

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp (hunk 5 above)
Comment (javi555) [reply modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp]: Done

File: modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h (hunk 4 above)
Comment (javi555) [reply modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h]: Done

File: modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h (hunk 3 above)
Comment (javi555) [reply modules/zmq_application/zmq_utils/include/zmq_middleware_facility.h:44]: Done

File: modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp [hunk 7]
```diff
@@ -5,25 +5,12 @@ ZmqMiddlewareFacility::ZmqMiddlewareFacility(zmq::context_t &context,
...
-    , _binaryFromZMQProvider(nullptr)
-    , _toZMQPublisher(nullptr)
-    , _subscriber(context, ZMQ_SUB)
-    , _publisher(context, ZMQ_PUB)
-    , _fromZmqMiddlewareProvider()
-    , _toZMQMiddlewareProvider(container, _publisher)
+                                             kpsr::Container *container, int numPublishers)
+    : _rbImgUrl(rbImgUrl), _rbWpUrl(rbWpUrl), _imgTopic(imgTopic), _wpTopic(wpTopic), _eventloop(eventloop), _poolSize(poolSize), _factory(factory), _numPublishers(numPublishers), _imageDataSubscriber(nullptr), _imageDataPublisher(nullptr), _binaryFromZMQProvider(nullptr), _subscriber(context, ZMQ_SUB), _fromZmqMiddlewareProvider()
 {
+  _publishers.reserve(_numPublishers);
```
Comment (pablogh-2000) [inline modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp]: ```_publishers(numPublishers)```

Comment (javi555) [reply modules/zmq_application/zmq_utils/src/zmq_middleware_facility.cpp]: Done

**Pull request merged**